stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -1,109 +1,147 @@
+import datetime as dt
+from _typeshed import Incomplete, ReadableBuffer, SupportsRead, SupportsReadline, SupportsTrunc, SupportsWrite, Unused
 from collections.abc import Callable, Iterable, Mapping, Sequence
 from types import TracebackType
-from typing import Any, TypeVar, overload
-from typing_extensions import Literal, Self, TypeAlias
+from typing import Any, NoReturn, Protocol, SupportsInt, TypeVar, overload, type_check_only
+from typing_extensions import Literal, Self, SupportsIndex, TypeAlias
 
-import psycopg2
-import psycopg2.extensions
 from psycopg2.sql import Composable
 
 _Vars: TypeAlias = Sequence[Any] | Mapping[str, Any] | None
+_AcceptedByInt: TypeAlias = str | ReadableBuffer | SupportsInt | SupportsIndex | SupportsTrunc
+
+@type_check_only
+class _type:
+    # The class doesn't exist at runtime but following attributes have type "psycopg2._psycopg.type"
+    name: str
+    values: tuple[int, ...]
+    def __call__(self, __value: str | bytes | None, __cur: cursor | None) -> Any: ...
+
+BINARY: _type
+BINARYARRAY: _type
+BOOLEAN: _type
+BOOLEANARRAY: _type
+BYTES: _type
+BYTESARRAY: _type
+CIDRARRAY: _type
+DATE: _type
+DATEARRAY: _type
+DATETIME: _type
+DATETIMEARRAY: _type
+DATETIMETZ: _type
+DATETIMETZARRAY: _type
+DECIMAL: _type
+DECIMALARRAY: _type
+FLOAT: _type
+FLOATARRAY: _type
+INETARRAY: _type
+INTEGER: _type
+INTEGERARRAY: _type
+INTERVAL: _type
+INTERVALARRAY: _type
+LONGINTEGER: _type
+LONGINTEGERARRAY: _type
+MACADDRARRAY: _type
+NUMBER: _type
+PYDATE: _type
+PYDATEARRAY: _type
+PYDATETIME: _type
+PYDATETIMEARRAY: _type
+PYDATETIMETZ: _type
+PYDATETIMETZARRAY: _type
+PYINTERVAL: _type
+PYINTERVALARRAY: _type
+PYTIME: _type
+PYTIMEARRAY: _type
+ROWID: _type
+ROWIDARRAY: _type
+STRING: _type
+STRINGARRAY: _type
+TIME: _type
+TIMEARRAY: _type
+UNICODE: _type
+UNICODEARRAY: _type
+UNKNOWN: _type
 
-BINARY: Any
-BINARYARRAY: Any
-BOOLEAN: Any
-BOOLEANARRAY: Any
-BYTES: Any
-BYTESARRAY: Any
-CIDRARRAY: Any
-DATE: Any
-DATEARRAY: Any
-DATETIME: Any
-DATETIMEARRAY: Any
-DATETIMETZ: Any
-DATETIMETZARRAY: Any
-DECIMAL: Any
-DECIMALARRAY: Any
-FLOAT: Any
-FLOATARRAY: Any
-INETARRAY: Any
-INTEGER: Any
-INTEGERARRAY: Any
-INTERVAL: Any
-INTERVALARRAY: Any
-LONGINTEGER: Any
-LONGINTEGERARRAY: Any
-MACADDRARRAY: Any
-NUMBER: Any
-PYDATE: Any
-PYDATEARRAY: Any
-PYDATETIME: Any
-PYDATETIMEARRAY: Any
-PYDATETIMETZ: Any
-PYDATETIMETZARRAY: Any
-PYINTERVAL: Any
-PYINTERVALARRAY: Any
-PYTIME: Any
-PYTIMEARRAY: Any
 REPLICATION_LOGICAL: int
 REPLICATION_PHYSICAL: int
-ROWID: Any
-ROWIDARRAY: Any
-STRING: Any
-STRINGARRAY: Any
-TIME: Any
-TIMEARRAY: Any
-UNICODE: Any
-UNICODEARRAY: Any
-UNKNOWN: Any
-adapters: dict[Any, Any]
+
+class _ISQLQuoteProto(Protocol):
+    # Objects conforming this protocol should implement a getquoted() and optionally a prepare() method.
+    # The real ISQLQuote class is implemented below with more stuff.
+    def getquoted(self) -> bytes: ...
```

Comment:
Nit: Could you add the commented out signature of `prepare()` as well to serve as documentation or for whenever we get optional protocol members? (python/typing#601)